src/software/new_geom/line.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <stdexcept>
+
+#include "software/new_geom/polynomial.h"
+
+class Line final : public Polynomial
+{
+   public:
+    /**
+     * Creates a line with zero y-intercept and zero slope
+     */
+    inline explicit Line() {}
+
+    /**
+     * Creates a line from a y-intercept and a slope
+     *
+     * @param y_intercept the y_intercept of the line
+     * @param slope the slope of the line
+     */
+    inline explicit Line(double y_intercept, double slope)
+        : Polynomial({y_intercept, slope})
+    {
+    }
+
+    /**
+     * Sets the coefficient of the term of the given order
+     *
+     * @param order the order of the term to set the coefficient
+     * @param coeff the coefficient
+     *
+     * @throws std::invalid_argument if order > 1
+     */
+    inline void setCoeff(unsigned int order, double coeff) override
+    {
+        if (order > 1)
+        {
+            throw std::invalid_argument(
+                "Tried to set the coefficient of term with order greater than 1");
```

Comment:
Might be good to indicate that we tried to do this for a ```Line``` and that its not allowed

Reply:
Addressed in 746b91a

Reply:
Hmm, does this violate the [Liskov Substitution Principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle)? I believe it's a violation for a subclass to accept a smaller set of arguments then it's super class, as this would mean that we could not substitute the subclass in-place of the super class in all cases. I also don't think that it's reasonable for a polynomial to be a subclass of a line (what does calling `getSlope()` mean for a polynomial?). So composition might be the way to go with this.

We should definitely chat about this today though!

Reply:
More understandable link: https://stackify.com/solid-design-liskov-substitution-principle/

Reply:
Yeah it does, didn't realize we were trying to stick to the principle. In that case, unless we make `Polynomial` immutable, `Line` can't be a true subtype of `Polynomial`.

Reply:
Hmm maybe we can have ```Line``` wrap a ```Polynomial``` with an order of 1 instead

Reply:
As in composition? Also private inheritance could work as well

Reply:
Addresed in 31c130a36c3b283ceeb3119c01c817db7f16bde2. I ended up choosing composition over private inheritance since StackOverflow suggests it should be preferred. I also thought that at this point `Line` being a standalone versus composition doesn't make much of a difference, but still maintaining `Polynomial` as the underlying `Line` representation seems to make more sense.

Reply:
AH yeah, in general it's good to follow the [SOLID](https://scotch.io/bar-talk/s-o-l-i-d-the-first-five-principles-of-object-oriented-design) principles for OOP stuff.